AutoCollection/HttpRequests.ts

```diff
@@ -90,26 +94,35 @@ class AutoCollectHttpRequests {
                         onRequest(request, response);
                     }
                 });
-            });
+            }
         }
 
-        var originalHttpsServer = https.createServer;
-        https.createServer = (options, onRequest) => {
-            return originalHttpsServer(options, (request:http.ServerRequest, response:http.ServerResponse) => {
-                // Set up correlation context
-                var requestParser = new HttpRequestParser(request);
-                var correlationContext = this._generateCorrelationContext(requestParser);
+        function wrapServerEventHandler(server) {
+            const originalOn = server.on.bind(server);
+            server.on = (eventType: string, eventHandler) => {
+                switch (eventType) {
+                    case 'request':
```

Comment:
Is the actual implementation case-sensitive? If not, we should compare against `eventType.toLowerCase()`

Reply:
These are the string mentioned in the official Node.js docs (https://nodejs.org/api/http.html#http_event_request) and as far as I know, they are kept as is inside the EventEmitter class. See: https://github.com/nodejs/node/blob/v8.4.0/lib/events.js#L232